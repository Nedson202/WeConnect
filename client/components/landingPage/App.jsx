import React, { Component } from 'react';
import Main from './Main';

/**
 * @class App
 *
 * @extends {Component}
 */
class App extends Component {
  /**
   * @description Set page document title   *
   * @returns {undefined}
   *
   * @memberof App
   */
  componentDidMount() {
    document.title = 'Welcome to WeConnect';
  }

  /**
   * @description Renders the component to the dom
   *
   * @returns {object} JSX object
   *
   * @memberof App
   */
  render() {
    return (
      <div>
        <div id="background-image" />
        <Main />
      </div>
    );
  }
}

export default App;
