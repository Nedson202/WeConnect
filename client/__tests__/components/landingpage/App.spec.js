import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import App from '../../../components/landingPage/App';

configure({ adapter: new Adapter() });

describe('Component: App', () => {
  beforeEach(() => {
    global.document = {
      title: () => {}
    }
  })

  it('should render', () => {
  const wrapper = shallow(<App />);
    expect(wrapper.find('div').length).toBe(2);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  }) 
});