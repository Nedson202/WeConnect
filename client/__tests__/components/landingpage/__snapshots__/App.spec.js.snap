// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should render 1`] = `
<div>
  <div
    id="background-image"
  />
  <Main />
</div>
`;
